src/libos/src/poll/do_epoll/epoll_file.rs

```diff
@@ -290,6 +290,15 @@ impl EpollFile {
     }
 }
 
+impl Drop for EpollFile {
+    fn drop(&mut self) {
+        let mut interest = self.interest.lock().unwrap();
+        for (_fd, entry) in interest.drain() {
+            entry.set_deleted();
```

Comment:
We may need unregister_observer after set_deleted

        let file = match entry.file() {
            Some(file) => file,
            // TODO: should we warn about it?
            None => return Ok(()),
        };
        file.unregister_observer(&(entry as _)).unwrap();